ledger-core/network/messagesender/service.go

```diff
@@ -78,6 +78,10 @@ func (dm *DefaultService) SendRole(ctx context.Context, msg payload.Marshaler, r
 		return throw.W(err, "failed to calculate role")
 	}
 
+	if nodes[0].Equal(dm.affinity.Me()) {
+		inslogger.FromContext(ctx).Debug("Send to myself")
```

Comment:
does this helps? if you are sending on the same machine, the log with trace will continue on the same machine

Reply:
Multiple messages might be send with the same traceid. It helps to me for debugging.